pkg/openstack/common.go

```diff
@@ -570,6 +587,12 @@ func (ed *EndpointDetail) CreateRoute(
 					}
 				}
 			}
+		} else if ed.Route.OverrideSpec.Spec != nil && ed.Route.OverrideSpec.Spec.TLS != nil {
+			if ed.Route.OverrideSpec.Spec.TLS.CACertificate == "" ||
+				ed.Route.OverrideSpec.Spec.TLS.Certificate == "" ||
+				ed.Route.OverrideSpec.Spec.TLS.Key == "" {
+				return ctrl.Result{}, fmt.Errorf("incomplete tls override data")
+			}
```

Comment:
also should then print a warning if someone adds both for one endpoint and mention that the cert secret content takes precedence. At least I think thats how it works right now.

Reply:
maybe we remove this check from this PR and you could submit a separate PR for doing those checks via the webhook?

Reply:
Yes, I tried it and when using secret with cert data, it has a privilege. We need to change the logic a little bit here if we remove the `ed.Route.OverrideSpec.Spec.TLS`  and check the provided CR with the webhooks, otherwise route cert will be looped in creation and deletion.